src/shim/sdk/DocumentSDK.ts

```diff
@@ -324,3 +325,32 @@ export function revokeAccess(documentID: string, revokeList: DocumentAccessList)
         .map(({message}) => message)
         .toPromise();
 }
+
+export const advanced = {
+    /**
+     * Decrypt the provided document given the edeks of the document and its data. Returns a Promise which will be resolved once the document has been successfully decrypted.
+     * @param {Uint8Array}  edeks        The encrypted deks for the documentData.
+     * @param {Uint8Array}  documentData Document data to decrypt
+     */
+    decryptUnmanaged: (edeks: Uint8Array, documentData: Uint8Array) => {
+        ShimUtils.checkSDKInitialized();
+        ShimUtils.validateEncryptedDeks(edeks);
```

Comment:
Since this can throw you probably want to catch it inside the `Future` for this function.

edit: I see the other validation functions aren't used this way, and I'm honestly not sure why. I generally don't expect a `Promise` based function to throw, and there's no `@throws` in the jsdoc that would make me think it would. @ernieturner is this intentional?

Reply:
Sort of, ya. We communicate Promise based failures with an error code that explains what went wrong with the operation and I didn't want to create error codes for invalid data being passed in. Looking back at it I don't feel super great about how it works, but I think we've probably just got to live with it at this point.